app/services/analytics.rb

```diff
@@ -1,16 +1,18 @@
 class Analytics
-  def initialize(user:, request:, sp:, first_path_visit_this_session:, ahoy: nil)
+  def initialize(user:, request:, sp:, session:, ahoy: nil)
     @user = user
     @request = request
     @sp = sp
     @ahoy = ahoy || Ahoy::Tracker.new(request: request)
-    @first_path_visit_this_session = first_path_visit_this_session
+    @session = session
   end
 
   def track_event(event, attributes = {})
+    update_session_events_and_paths_visited_for_analytics(event) unless attributes[:success]==false
```

Comment:
Nit: Spacing:

```suggestion
    update_session_events_and_paths_visited_for_analytics(event) unless attributes[:success] == false
```

Since that exceeds the 100 character limit, maybe instead:

```suggestion
    update_session_events_and_paths_visited_for_analytics(event) if attributes[:success] != false
```